warehouse/query-core/src/main/java/datawave/query/jexl/functions/QueryFunctionsDescriptor.java

```diff
@@ -138,12 +142,63 @@ public Set<String> fieldsForNormalization(MetadataHelper helper, Set<String> dat
 
         @Override
         public Set<String> fields(MetadataHelper helper, Set<String> datatypeFilter) {
-            return JexlASTHelper.getIdentifierNames(args.get(0));
+            Set<String> fields = new HashSet<>();
+            switch (name) {
+                case QueryFunctions.COUNT:
+                case QueryFunctions.SUM:
+                case QueryFunctions.MIN:
+                case QueryFunctions.MAX:
+                case QueryFunctions.AVERAGE:
+                case QueryFunctions.GROUPBY_FUNCTION:
+                case QueryFunctions.NO_EXPANSION:
+                case QueryFunctions.LENIENT_FIELDS_FUNCTION:
+                case QueryFunctions.STRICT_FIELDS_FUNCTION:
+                    // In practice each of these functions should be parsed from the query
+                    // almost immediately. This implementation is added for consistency
+                    for (JexlNode arg : args) {
+                        fields.addAll(JexlASTHelper.getIdentifierNames(arg));
```

Comment:
Just for clarification - does it matter that we're doing a simple addAll(identifier) here but down below we're doing a UniqueFields.from(identifier)?

Reply:
👍